cpu/arm7_common/include/irq_arch.h

```diff
@@ -37,20 +44,23 @@ unsigned irq_disable(void)
     return _cpsr;
 }
 
-unsigned irq_restore(unsigned oldCPSR)
+static inline __attribute__((always_inline)) void irq_restore(unsigned oldCPSR)
 {
-    unsigned _cpsr;
-
-    _cpsr = __get_cpsr();
-    __set_cpsr((_cpsr & ~IRQ_MASK) | (oldCPSR & IRQ_MASK));
-    return _cpsr;
+    __set_cpsr(oldCPSR);
 }
 
-unsigned irq_enable(void)
+static inline __attribute__((always_inline)) unsigned irq_enable(void)
 {
     unsigned _cpsr;
 
     _cpsr = __get_cpsr();
     __set_cpsr(_cpsr & ~IRQ_MASK);
     return _cpsr;
 }
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* IRQ_ARCH_H */
+/** @} */
```

Comment:
Missing matching `@{` doxygen comment. You could also add some boilerplate Doxygen comment above, so that these function show up in the right group.